src/linear_codes/mod.rs

```diff
@@ -113,15 +116,21 @@ where
 
         let mat = Matrix::new_from_flat(n_rows, n_cols, &coeffs);
 
-        // 2. Apply Reed-Solomon encoding row-wise
+        // 2. Apply encoding row-wise
         let ext_mat =
             Matrix::new_from_rows(mat.rows().iter().map(|r| Self::encode(r, param)).collect());
 
         (mat, ext_mat)
     }
 
-    /// Tensor the point
-    fn tensor(point: &P::Point, left_len: usize, right_len: usize) -> (Vec<F>, Vec<F>);
+    /// Tensor the point.
```

Comment:
This function should not be called `tensor` - it has nothing to do with that in the univariate case. Also, I don't think the comment should prescribe what the function has to do: "For a univariate polynomial it returns a tuple,...". That's up to the implementor and it depends on how the matrix is constructed. If anything, we could mention those as examples:
"For instance, in the univariate Ligero case, this returns a tuple
  ((1, z, z^2, ..., z^n), (1, z^n, z^(2n), ..., z^((m-1)n)))"
Still, a trait method description giving the example of a specific implementor feels odd.

As an additional quibble, as I already mentioned, multilinar and univariate are not the only cases this trait could refer to.

Reply:
The naming follows `10.5.1 Identifying Tensor Product Structure in Polynomial Evaluation Queries` from Thaler's book.
I agree though that the doc comment should not specify examples. I'll move the uni/ml cases to their implementors

Reply:
How about being very explicit:
```
/// Tensor the query point z in the following sense:
/// For a polynomial p(X) = \sum_{k=0}^{n*m-1} p_k X^k,
/// represented by a matrix M with n rows and m columns
/// such that M_{i,j} = p_{i + n*j}, we define the tensoring of `z`
/// (a, b) = tensor(z, n, m) such that:
/// p(z) = b^T.M.a
/// returns the evaluation of p at z.
fn tensor(z: &P::Point, n: usize, m: usize) -> (Vec<F>, Vec<F>);
```

Reply:
You're absolutely right about the naming - I had no idea some reputable sources also call this a tensor in the univariate case, sorry. Let's leave the name as-is, then.

As for the explanation, it now refers only to the univariate case. But I think what remains if you replace the first equation by simply "p(x)" is general and accurately describes what this should do.

Reply:
Oh yes, good point, I meant to make the thing general and then forgot that p(x) in this definition is for univariate only